cirq/ops/moment.py

```diff
@@ -134,6 +134,16 @@ def operation_at(self, qubit: raw_types.Qid) -> Optional['cirq.Operation']:
         else:
             return None
 
+    def single_gate_in_moment(self) -> Optional['cirq.Gate']:
+        target = None
+        if not self.operations:
+            return None
+        target = self.operations[0].gate
```

Comment:
The operation might not be a `GateOperation`, thus it might not have a gate attribute. Can you write a test for that and check for it here? Thanks!

Reply:
`gate` is a property of the base `Operation` class: https://github.com/quantumlib/Cirq/blob/b2f8d43ad365bdd5b4c89c8999f310732f619838/cirq-core/cirq/ops/raw_types.py#L390-L392

I have #3678 open to suggest moving away from this, but no movement there.

Reply:
Ahh.